### ISSUE TITLE
Regenerative Fighter/Missle Bay Outfits

### ISSUE THREAD

**Issue by Hondolor** [endless-sky/endless-sky#1249]
This concept has been mentioned before but this idea has a twist..

The Regenerative Bay's have a high Outfit Cost and are slow to refill compared to buying at an outfitter, but give you unlimited Fighters/missles if your patient enough. Just throwing this out there

**Comment by Alkallid** [comment]
Do you mean in-game time or days?

**Comment by mdsmestad** [comment]
@Alkallid Yes

**Comment by XHawk87** [comment]
@Alkallid I don't believe that was a yes/no question.

Do you mean they regenerate while you're sitting around in a system or do they only regenerate between hyperspace/jumps/planet launches?

I don't personally use missiles or fighters due to their repeated cost on use. Its far simpler to not have to think about costs when firing weapons and stick to something that is free to fire. Beam weapons have always sufficed for my needs, so I don't tend to use anything else. Perhaps something like this would make them worth using.

**Comment by LocalGod79** [comment]
i think he means that it would be nice to have as an example i considered some months ago, nuclear type reactor cores spit out depleted uranium pellets (similarly to fuel and ramscoops) that could then be used as ammunition in a gun.

or just generally make it so that outfits can generate *and* consume a non-energy/heat attributes (which might also have a corresponding mass/capacity).

i think that coding this might be pretty difficult, since the variables need to be defined before the outfits try to...

**Comment by Alkallid** [comment]
Isn't already in place for names of ammo, in the capacity attribute?

**Comment by LocalGod79** [comment]
indeed, capacities and consumption are already supported, but "generation" is another story entirely, since rates need to be defined, and added to the clock...

i always just used "firing fuel" to limit how many shots a player could fire, with the potential to regenerate their firing ability granted through ramscoops.

**Comment by toilethinges** [comment]
@LocalGod79, they might not be outfits that sell for anything at all - however, I agree that if you could fabricate meteor missiles, torpedoes, or other comparable ammunitions, that would be an extremely unbalanced source of income.

(Although it would be really nice to have a Minelayer fabricator, because it's hard to get ammunition for those.)

**Comment by onnikyla** [comment]
it would make sense to have fabricators for mines, it just makes sense for the automatons to generate everything needed without landing on a planet.

**Comment by OmegaZanicus** [comment]
@onnikyla Here again, generating outfits could be exploited, whether it makes sense to have it available or not. The only solutions to these problems would either be to implement an attribute to prevent it from being sold, or to make it a zero cost item.

**Comment by LocalGod79** [comment]
@OmegaZanicus in which case, just have the outfit use hyperspace fuel...

**Comment by mdsmestad** [comment]
@toilethinges I disagree. If it's producing slow then it's not a very big issue. Although I still think re-sell value in all outfits should be docked

**Comment by Pointedstick** [comment]
I'm re-opening this since it's a credible option to fix the issues raised in https://github.com/endless-sky/endless-sky/issues/297, https://github.com/endless-sky/endless-sky/issues/906, https://github.com/endless-sky/endless-sky/issues/952, https://github.com/endless-sky/endless-sky/issues/1932, https://github.com/endless-sky/endless-sky/issues/3920, and https://github.com/endless-sky/endless-sky/issues/4034.

In order to maintain balance, it would probably require restrictions like the following:
- Moderately large crew, space, and energy requirements
- Long regen time time so that you couldn't realistically have infinite ammunition/fighter capacity during a battle (e.g. multiple real-time minutes or in-game days per fighter or missile)
- Requires consumable commodities (e.g. Metal, Electronics, Heavy Metals), purchased from the trading window or looted from enemy ships
- Require special "blueprints" outfits that detail how to manufacture certain items. These could be easily purchasable for common human weapons, but should be gated behind missions for more exotic alien missiles and fighters. You should have to work hard to reach the point where you have regenerating Korath drones armed with Piercers!

**Comment by KyroFirehammer** [comment]
If they are to cost commodities, which is a great idea, they should also be able to pull from the resources on othet ships in the fleet because most of the time carriers cargo capacity is fairly limited. As balance carriers could possibly only be able to pull from other ship resourced when on a planet (or not in combat if they work while not landed)

**Comment by Hacklin** [comment]
A Bullet Machine Shop to create `Gatling Ammo` is probably fine.
They just convert `Industrials` + `Energy` + `Credits` into `Gatling Ammo` + `Garbage` (commodity).
By having a fab consume credits you can prevent a player gaming the system money-wise;
i.e. it should be more expensive to produce an ordnance on a ship than buying it in an outfitter.

A fab/assembly to create missiles, mines or torpedoes runs counter to lore.
For example, where do you get those advanced alien electronics, sensors, alloys and warheads from? Just build them out of commodities? On a ship?
That said, a Javelin Workshop to create `Javelins` is probably fine.

Fighters and drones are miniature spaceships.
All the problems you have with missiles/torpedoes also holds for fighters and drones.
Plus, fighters and drones contain weapons, engines, power plants and other outfits that can be used separately on other spaceships.

The more sophisticated an ordnance is the more high-tech parts you need, the larger the factory will be. Therefor IMO, any fabs should only be allowed on world ships and special factory ships
with the exception for very simple ordnance like bullets and Javelins.

What you want is:
1. the ability to fire more missiles/torpedoes, lay more mines and launch more fighters/drones and
2. not to have to fly al the way to an outfitter that sells them in order to replenish them.

For missiles, mines and torpedoes you can solve:
1. by installing storage racks to increase the amount of ordnance a ship carries and
2. by buying missiles, mines or torpedoes in bulk and loading them into the cargo hold of freighters and other support ships. (If you want to have a large fleet, you've got to deal with the added logistics of supply ships.)

For fighters and drones you can solve:
1. by adding Hangars to carriers and putting a Reload time on the Fighter and Drone Bays (note that the Reload time could also be used when retrieving the fighters and drones) and
2. by allowing to store fighters [without crew] and drones in the cargo hold of freighters.

**Comment by Ferociousfeind** [comment]
I don't think fabricating fighters after a battle and selling them would break anything. After all, you can sit around and capture pirate ships for ridiculous amounts of credits. Simply calculate the amount of money you reasonably gain per real-time minute, and balance regeneration of certain outfits or entire ships against that.

For example, say you can make an even 1,000,000 credits in an hour of real time, capturing Sparrows and Clippers and the like, with your fancy Bactrian and a crew of 245. This gives you around 4080 credits per hour per crewmember. So, if you wanted to keep the balance relatively even, perhaps underpowered due to the lack of real input from the player, make a gatling ammo manufacturing plant require 2-3 crew, and create about 30 ammo per minute, or one every two seconds.

I honestly like the idea of regenerating ammunition, and regenerating fighters/drones. That's my input, anyways.

**Comment by Amazinite** [comment]
We don't think that outfits that generate fighters or missiles (or other outfits) are fitting for the game. This would effectively introduce a form of crafting to the game, which is definitely in the territory of features creep as described by https://github.com/endless-sky/endless-sky/issues/1408#issuecomment-238399248.

While this could be seen as a potential solution to the problems that people have with fighters, I don't think that this would actually be the best solution. As discussed (somewhere) in #4034 (or maybe it was one of the other hundred fighter issues), a large draw of the game is the ability to customize each and every one of your ships, fighters and drones included. By creating outfits that produce fighters, how would these newly created fighters behave? Would they have some stock loadout, somewhat defeating the purpose of being able to customize your fighters, or would they remember what outfits they had before they were destroyed, which would add a lot of complexity to this system?

I'd rather that instead of making fighters easier to replace, we made them harder to lose with mechanics such as the one I described in https://github.com/endless-sky/endless-sky/pull/8989#issuecomment-1627818437 or #9760.
